tests/src/Unit/ShouldUpdateTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Drupal\Tests\updates_log\Unit;
+
+
+/**
+ * @coversDefaultClass \Drupal\updates_log\UpdatesLog
+ * @group updates_log
+ */
+class ShouldUpdateTest extends UpdatesLogTestBase {
+
+  /**
+   * @covers ::ShouldUpdate
+   */
+  public function testFirstTime(): void {
+    $status = $this->updatesLog->shouldUpdate(time(), NULL);
+    $this->assertTrue($status);
+  }
+
+  /**
+   * @covers ::ShouldUpdate
+   */
+  public function testImmediately(): void {
+
+    $status = $this->updatesLog->shouldUpdate(time(), time() + 1);
+    $this->assertFalse($status);
+  }
+
+  /**
+   * @covers ::ShouldUpdate
+   */
+  public function testLater(): void {
+    $status = $this->updatesLog->shouldUpdate(time(), time() - 24 * 60 * 60);
+    $this->assertTrue($status);
+  }
+
+  /**
+   * @covers ::ShouldUpdate
+   */
+  public function testTestEnv(): void {
+    putenv('UPDATES_LOG_TEST=1');
```

Comment:
unrelated.
we could consider setting up the project for Silta too:
* to run code scanners
* to run tests